### ISSUE TITLE
Missing focus trap for responsive sidebar navigation

### ISSUE THREAD

**Issue by JoseeWouters** [tailwindlabs/tailwindui-issues#620]
**What component (if applicable)**
- URL for category: https://tailwindui.com/components/application-ui/page-examples/settings-screens
- Component name: With sidebar navigation and constrained content area

**Describe the bug**
When tabbing through the example with the collapsible sidebar placed "above" the content, you can continu tabbing through the underlying content, instead of staying in the sidebar.

**To Reproduce**
1. Go to the example with sidebar navigation and constrained content area
2. Resize the example to see the responsive version with collapsible sidebar
3. Tab through the sidebar
4. After the last item "logout" the focus disappears to the underlying content

**Expected behavior**
I expect the focus to be trapped in the sidebar.

**Screenshots**
<img width="721" alt="afbeelding" src="https://user-images.githubusercontent.com/12784125/118804095-888cac80-b8a4-11eb-8983-78bfa6ca8871.png">

**Browser/Device (if applicable)**
 - OS: MacOS Catalina
 - Browser: Chrome, Firefox
 - Version: most recent

**Additional context**

**Comment by JoseeWouters** [comment]
I solved it by adding a `aria-modal="true"` on the first real `div` in this example: 
`<div className="relative max-w-xs w-full bg-white pt-5 pb-4 flex-1 flex flex-col" aria-modal="true">`.

I would be open to creating a PR to fix this, but I couldn't find the correct place in the https://github.com/tailwindlabs/headlessui repo to do this.
